docs/UserGuide.md

```diff
@@ -266,44 +309,47 @@ Examples:
+Users can specify additional optional parameters to limit the keyword matching to the fields that is associated with each
+flag. Parameters will only match results that contain the associated field (E.g. n/Hans will not return any `activities`
 since `activities` do not have a `name` field.)
 
-The flags are:
+Format: `find [OPTIONAL_PARAMETERS]...`
+
+The parameters are:
+
 * `n/NAME`: Matches the name field (Valid for: `Patients`, `Doctors`)
 * `c/CONTACT_NUMBER`: Matches the contact number field (Valid for: `Patients`, `Doctors`)
 * `a/AGE`: Matches the age field (Valid for: `Patients`)
 * `g/GENDER`: Matches the gender field; The only valid keywords for this field are `M/F/O` (Valid for: `Patients`)
 * `h/HEIGHT`: Matches the height field (Valid for: `Patients`)
 * `w/WEIGHT`: Matches the weight field (Valid for: `Patients`)
-* `b/BLOOD_TYPE`: Matches the blood type field; The only valid keywords for this field are `A/B/AB/O` 
-(Valid for: `Patients`)
-* `m/MEDICAL_CONDITION`: Limits the keyword search to the list of medical conditions of a patient (Valid for: `Patients`)
+* `b/BLOOD_TYPE`: Matches the blood type field; The only valid keywords for this field are `A/B/AB/O`
+  (Valid for: `Patients`)
+* `m/MEDICAL_CONDITION`: Limits the keyword search to the list of medical conditions of a patient (Valid
+  for: `Patients`)
 * `s/DEPARTMENT`: Matches the department field (Valid for: `Doctors`)
 * `ti/TITLE`: Matches the title field (Valid for: `Activities`)
-* `ta/TAG_DESCRIPTION`: Matches results that contain the specified tag in its list of tags (Valid for: `Activities`, 
-_Tagging for `Doctors` and `Patients` coming soon_)
-
-Format: `find [OPTIONAL_FLAG/]KEYWORD...`
-
-* Keyword is case-insensitive for convenience (“dia” will match diabetic patients even if the user stored the 
-patient's condition as “Diabetes”)
-* Flags can be repeated (e.g. `find n/Hans n/Bo` will return both `Hans Gruber` and `Bo Yang`)
-* If the optional flag is not specified, the keyword will match any fields.
-E.g. `find dia` will return:
+* `ta/TAG_DESCRIPTION`: Matches results that contain the specified tag in its list of tags (Valid for: `Activities`,
+  _Tagging for `Doctors` and `Patients` coming soon_)
+
+Note: 
+* Keyword is case-insensitive for convenience (“dia” will match diabetic patients even if the user stored the patient's
+  condition as “Diabetes”)
+* Parameters can be repeated (e.g. `find n/Hans n/Bo` will return both `Hans Gruber` and `Bo Yang`)
+* If the optional parameters is not specified, the keyword will match any fields. E.g. `find dia` will return:
     1. Doctor Claudia, whose name matches `dia`
     2. Patient Jaryl, whose medical condition, `diabetes`, matches `dia`
     3. Doctor Tom, whose specialty, `Pediatrics`, matches `dia`
     4. Patient Lydia, whose name matches `dia`
 
 Examples:
+
 * `find m/diabetes a/42 n/Jaryl`
```

Comment:
Likewise, if the previous suggestion for the parameter flag of patients' medical conditions is changed, then we can update this example as well.

```suggestion
* `find o/diabetes a/42 n/Jaryl`
```

Reply:
yup sure